ports/geckolib/wrapper.rs

```diff
@@ -342,47 +339,32 @@ impl<'ln> TNode for GeckoNode<'ln> {
 }
 
 #[derive(Clone, Copy)]
-pub struct GeckoDocument<'ld> {
-    document: *mut RawGeckoDocument,
-    chain: PhantomData<&'ld ()>,
-}
-
-impl<'ld> GeckoDocument<'ld> {
-    pub unsafe fn from_raw(doc: *mut RawGeckoDocument) -> GeckoDocument<'ld> {
-        GeckoDocument {
-            document: doc,
-            chain: PhantomData,
-        }
-    }
-}
+pub struct GeckoDocument<'ld>(pub &'ld RawGeckoDocument);
 
 impl<'ld> TDocument for GeckoDocument<'ld> {
     type ConcreteNode = GeckoNode<'ld>;
     type ConcreteElement = GeckoElement<'ld>;
 
     fn as_node(&self) -> GeckoNode<'ld> {
-        unsafe { GeckoNode::from_raw(self.document as *mut RawGeckoNode) }
+        unsafe { GeckoNode(&*(self.0 as *const _ as *const RawGeckoNode)) }
     }
 
     fn root_node(&self) -> Option<GeckoNode<'ld>> {
         unsafe {
-            Gecko_GetDocumentElement(self.document).as_ref().map(|el| GeckoElement::from_ref(el).as_node())
+            Gecko_GetDocumentElement(self.0).borrow_opt().map(|el| GeckoElement(el).as_node())
         }
     }
 
     fn drain_modified_elements(&self) -> Vec<(GeckoElement<'ld>, GeckoElementSnapshot)> {
         unimplemented!()
         /*
-        let elements =  unsafe { self.document.drain_modified_elements() };
+        let elements =  unsafe { self.0.drain_modified_elements() };
         elements.into_iter().map(|(el, snapshot)| (ServoLayoutElement::from_layout_js(el), snapshot)).collect()*/
     }
 }
 
 #[derive(Clone, Copy)]
-pub struct GeckoElement<'le> {
-    element: *mut RawGeckoElement,
-    chain: PhantomData<&'le ()>,
-}
+pub struct GeckoElement<'le>(pub &'le RawGeckoElement);
```

Comment:
I'd sort of prefer the `.element` field name to make it similar to servo's implementation (here and in `Node`/`Document`. But I guess I'm fine with this too.